RxSwift/Subjects/Variable.swift

```diff
@@ -10,7 +10,7 @@
 ///
 /// Unlike `BehaviorSubject` it can't terminate with error, and when variable is deallocated
 /// it will complete it's observable sequence (`asObservable`).
-public final class Variable<Element> {
+public final class Variable<Element>: ObservableConvertibleType {
```

Comment:
Hi @tarunon ,

the mere fact that we are relying on the fact will somebody retain variable or not means that this is fragile API (the part about completing variable observable sequence on deinit). That's why I don't like it.

It is pretty normal to create a reference to extension source, being it `ObservableType` or `ObservableConvertibleType`. This is being done all the time, and it shouldn't be a problem.

That's why `Variable` doesn't conform to `ObservableConvertibleType`, because that increases the chances of somebody retaining it.

Every solution has it's drawbacks.